plugins/module_utils/redfish_utils.py

```diff
@@ -13,6 +13,11 @@
 from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
 from ansible.module_utils.six.moves.urllib.parse import urlparse
 
+import os
+from pathlib import Path
+
+HOME = str(Path.home())
```

Comment:
Is this something we want to put on as a requirement to the user? Why not have this as a parameter?

Reply:
@felixfontein do other Ansible modules use files like this to pass information so they don't need to be entered as arguments? If so, can you please point me to an example so I can get a better sense of what the best practice is for doing that?

Reply:
Yes, we can follow the standard practice of storing the sessionfile in whatever directory.  Please let us know.  
Please note that session management is being introduced here for ansible to avoid login/logout multiple times for each operation.

Reply:
> @felixfontein do other Ansible modules use files like this to pass information so they don't need to be entered as arguments? If so, can you please point me to an example so I can get a better sense of what the best practice is for doing that? What's also important here is the intent is to pass a session token; are there other cases in Ansible where such data is stored so it doesn't need to be passed as a parameter?

So far I've only seen this for modules which correspond in functionality to CLI tools, and where the CLI tools read such configurations. One such example is #3250.

I don't think it is a good idea to encode something that looks very specific (`"~/sessionfile" + self.root_uri[8:] + ".txt"`) in generic vendor-agnostic code like redfish_utils.py. I think it would be better to add some functionality which allows to provide this session token to `RedfishUtils`, so that a vendor-specific module can use that to read the token from a file and provide it to the shared code. Alternatively, other modules could also allow to provide the session ID by a module option, instead of reading it from a file. Then users can pass the session ID in also from Ansible Vault, a file lookup, or whatever other source they have.

Reply:
That's perfect; we'll need to look at that more. At least the concept of a session token is standard in Redfish, but the way it's being passed around here is vendor-specific.

Thanks for the pointer @felixfontein ! @rajeevarakkal let's look into using the Ansible Vault for storing/fetching the session token so we don't have to do this file processing here.

Reply:
@mraineri modules cannot access Ansible Vault; the user needs to pass the session ID from vault to the module. So to support this, you need a module option which allows to pass the session ID in.

Reply:
@mraineri , @felixfontein , We can always improve on our method of caching session id.  For now, we have adopted txt file and it is working.  Also, it does not break the existing functionality.  We can look at Vault or any other way in future iterations.

Reply:
@mraineri , we have completed all the changes and tested it out as well.  Any enhancements we can take it in the future w.r.t this feature.